libpod/kube.go

```diff
@@ -0,0 +1,277 @@
+)
+
+// InspectforKube takes a slice of libpod containers and generates
+// one v1.Pod description that includes just a single container.
+func InspectForKube(containers []*Container) error {
+	var output []byte
+	developmentComment := []byte("# Generation of Kubenetes YAML is still under development!\n")
+	logrus.Warn("This function is still under heavy development.")
+	// Iterate each container
+	for _, ctr := range containers {
+		// Generate the v1.Pod yaml description
+		pod, err := simplePodWithV1Container(ctr)
+		if err != nil {
+			return err
+		}
+		// Marshall the results
+		b, err := yaml.Marshal(pod)
+		if err != nil {
+			return err
+		}
+		output = append(output, developmentComment...)
+		output = append(output, b...)
+		// Output the v1.Pod with the v1.Container
+		fmt.Println(string(output))
+	}
+	return nil
+}
+
+// simplePodWithV1Container is a function used by inspect when kube yaml needs to be generated
+// for a single container.  we "insert" that container description in a pod.
+func simplePodWithV1Container(ctr *Container) (*v1.Pod, error) {
+	var containers []v1.Container
+	result, err := ContainerToV1Container(ctr)
+	if err != nil {
+		return nil, err
+	}
+	containers = append(containers, result)
+
+	tm := v12.TypeMeta{
+		Kind:       "Pod",
+		APIVersion: "v1",
+	}
+
+	// Add a label called "app" with the containers name as a value
+	labels := make(map[string]string)
+	labels["app"] = removeUnderscores(ctr.Name())
+	om := v12.ObjectMeta{
+		// The name of the pod is container_name-libpod
+		Name:   fmt.Sprintf("%s-libpod", removeUnderscores(ctr.Name())),
+		Labels: labels,
+		// Seems required so added this, else shows null
+		CreationTimestamp: v12.Now(),
+	}
+	ps := v1.PodSpec{
+		Containers: containers,
+	}
+	p := v1.Pod{
+		TypeMeta:   tm,
+		ObjectMeta: om,
+		Spec:       ps,
+	}
+	return &p, nil
+}
+
+// ContainerToV1Container converts information we know about a libpod container
+// to a V1.Container specification.
+func ContainerToV1Container(c *Container) (v1.Container, error) {
+	container := v1.Container{}
+	kubeSec, err := generateKubeSecurityContext(c)
+	if err != nil {
+		return container, err
+	}
+
+	if len(c.Spec().Linux.Devices) > 0 {
+		// TODO Enable when we can support devices and their names
+		devices, err := generateKubeVolumeDeviceFromLinuxDevice(c.Spec().Linux.Devices)
+		if err != nil {
+			return container, err
+		}
+		container.VolumeDevices = devices
+		return container, errors.Wrapf(ErrKubeNotSupported, "linux devices")
+	}
+
+	if len(c.config.UserVolumes) > 0 {
+		// TODO When we until we can resolve what the volume name should be, this is disabled
+		volumes, err := libpodMountsToKubeVolumeMounts(c)
+		if err != nil {
+			return container, err
+		}
+		container.VolumeMounts = volumes
+		return container, errors.Wrapf(ErrKubeNotSupported, "volume names")
+	}
+
+	containerCommands := c.Command()
+	container.Name = removeUnderscores(c.Name())
```

Comment:
Can we rename this from "container" to something a bit clearer? kubeContainer?

Reply:
done